odoo/addons/test_assetsbundle/tests/test_assetsbundle.py

```diff
@@ -1925,3 +1891,137 @@ def test_37_path_can_be_an_attachment(self):
              .my_div.subdiv{color: blue;}
             """
         )
+
+@tagged('-at_install', 'post_install')
+class AssetsNodeOrmCacheUsage(TransactionCase):
+
+    def cache_keys(self):
+        keys = self.env.registry._Registry__cache.d
+        asset_keys = [key for key in keys if key[0] == 'ir.asset' and '_get_asset_paths' in str(key[1])] # ignore topological sort entry
+        qweb_keys = [key for key in keys if key[0] == 'ir.qweb']
+        return asset_keys, qweb_keys
+
+    def test_assets_node_orm_cache_usage_debug(self):
+        self.env.registry._clear_cache()
+
+        asset_keys, qweb_keys = self.cache_keys()
+        self.assertEqual(len(asset_keys), 0)
+        self.assertEqual(len(qweb_keys), 0)
```

Comment:
you can refactor this function and all the tests to
`assert_cache_keys(expected_number_assets_keys, expected_number_qweb_keys)`  --> you will remove 2 lines of copy/pasted call for each assert

Reply:
would seem over-engineered to me, the current way is very readable